flow/layers/layer.h

```diff
@@ -49,6 +49,7 @@ struct PrerollContext {
   MutatorsStack& mutators_stack;
   SkColorSpace* dst_color_space;
   SkRect cull_rect;
+  bool layer_reads_from_surface;
```

Comment:
This could use a better name, but I don't like "skia layer needs readback" because skia doesn't have needs, they provide support. I'll think of something better that reflects that this is the current status of whether or not we've encountered an engine layer that uses a read-back operation...

"readback_operation_detected"?

Still thinking...

Reply:
I renamed it to subtree_performs_readback_operation and created an AutoPrerollSaveLayer class to manage the flags for saveLayer containers.

Reply:
Hmm... it's still unclear to me what's the root of the subtree. At a first glance, I'd think the root to be the current layer that's doing preroll. But it's possible that the subtree of the current layer doesn't do readback, back the `subtree_performs_readback_operation` is still true because its older sibling performs readback.

How about `needs_readback_from_current_skia_layer`? Then everything seems to be more intuitive: whenever Skia saveLayer is called, we're creating a new Skia layer, so we clear this flag.

Reply:
The root of the subtree is basically since the last saveLayer in this case. It's really a property for the current drawable surface. How about "readback_detected_on_current_drawable" or something?

You keep suggesting names containing "skia", but I feel that the name of the graphics library is an implementation detail. It's really that the "thing" we are rendering onto is a drawable or surface and the fact that we get them from Skia is incidental to the property being tracked.

Reply:
=> surface_needs_readback